### PR TITLE
[DATA-837] Changes for implement CreatePackage in app Package Service

### PR DESCRIPTION
**Pull request by tahiyasalam**
`PackageInfo` will be sent in the upload request by the user
`UploadedPackage` will be saved in the DB 
`Package` will be returned to the user in the GetPackage(s) response(s) 

Associated `app` changes are [here](https://github.com/viamrobotics/app/pull/1009)

### CODE REVIEW

File: proto/viam/app/packages/v1/packages.proto [hunk 1]
```diff
@@ -56,7 +57,6 @@ message PackageInfo {
   PackageType type = 4;
   repeated FileInfo files = 5;
   google.protobuf.Struct metadata = 6;
-  google.protobuf.Timestamp created_on = 7;
```
Comment (agreenb) [inline proto/viam/app/packages/v1/packages.proto:59]: I would remove this in a separate PR since it's a breaking change.

**Review by agreenb — CHANGES_REQUESTED** [review]
Looks great, and agree that it's better to embed the fields directly in the UploadedPackage.

File: proto/viam/app/packages/v1/packages.proto [hunk 2]
```diff
@@ -83,6 +83,18 @@ message DeletePackageResponse {
 message Package {
   PackageInfo info = 1;
   string uri = 2;
+  google.protobuf.Timestamp created_on = 3;
+}
+
+message UploadedPackage {
```
Comment (agreenb) [inline proto/viam/app/packages/v1/packages.proto:89]: Can you add a comment that this is what's stored in the DB?

File: proto/viam/app/packages/v1/packages.proto (hunk 1 above)
Comment (tahiyasalam) [reply proto/viam/app/packages/v1/packages.proto]: Yup, added a ticket as a reminder

File: proto/viam/app/packages/v1/packages.proto (hunk 2 above)
Comment (AaronCasas) [reply proto/viam/app/packages/v1/packages.proto:91]: If the difference between Package and UploadedPackage is essentially that UploadedPackage is our internal view, I wonder if it should have a name indicating that. Might be worth asking Eric if there's a generally accepted pattern to follow here? Non blocking, but worth checking imo

Comment (tahiyasalam) [reply proto/viam/app/packages/v1/packages.proto:91]: done!

Comment (agreenb) [reply proto/viam/app/packages/v1/packages.proto:91]: +1 I think InternalPackage would be a good rename, and include that wording of "our internal view" in the comment.

We'll include Eric in the force-merge to remove created_on so can additionally get input there — want this merge to be unblocked for now.

Comment (tahiyasalam) [reply proto/viam/app/packages/v1/packages.proto]: updated

**Review by agreenb — APPROVED** [review]
Looks great, thanks for making the updates!

**Pull request merged**